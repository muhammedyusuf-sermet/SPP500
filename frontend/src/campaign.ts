export interface ICampaignState {
	Id?: string,
	Name: string,
	Description: string;
	// Monsters: Monster.IMonsterState[];
}
