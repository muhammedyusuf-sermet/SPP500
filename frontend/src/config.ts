export const API_URL = "https://http://3.18.65.138:3000";