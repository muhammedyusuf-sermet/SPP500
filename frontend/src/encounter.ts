import { IMonsterData } from "./monster";

export interface IEncounterData {
	Id?: string,
	Name: string,
	Description: string;
	Monsters: IMonsterData[];
}
