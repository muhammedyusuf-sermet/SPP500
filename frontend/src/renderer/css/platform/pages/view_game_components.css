.view-game-components-container {
	text-align: center;
}
.page-title {
	padding: 5px 15px;
	display: inline-block;
}

.view-game-components-container .tab-root {
	flex-grow: 1;
}

.view-game-components-container .app-bar {
	position: relative;
}

.view-game-components-container .layout {
	width: auto;
	margin-left: 2%;
	margin-right: 2%;
}

.view-game-components-container .card-grid {
	padding: 8px 0;
}

.view-game-components-container .card {
	height: 100%;
	display: flex;
	flex-direction: column;
}

.view-game-components-container .card-media {
	padding-top: 56.25%; /* 16:9 */
}

.view-game-components-container .card-content {
	flex-grow: 1;
}
