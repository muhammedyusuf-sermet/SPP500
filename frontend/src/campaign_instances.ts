import * as Campaign from "./campaign";

// Dummy array of monsters
import EncounterInstances from './encounter_instances';

let encounterExample: Campaign.ICampaignState;
encounterExample = {
	Id: 0,
	Name: "Campaign",
	Summary: "Bloody hell...",
	Notes: "You gonna die!",
	Encounters: EncounterInstances['EncounterInstances'],
};

let CampaignInstances = new Array();

// Generate a dummy array of encounters with different IDs
for (var i = 0; i < 12; i++) {
	let encounter = {} as Campaign.ICampaignState;
	Object.assign(encounter, encounterExample);
	encounter['Id'] = i;
	CampaignInstances.push(encounter);
}

export default {
    CampaignInstances,
}