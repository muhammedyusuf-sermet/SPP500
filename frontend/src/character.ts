import { ICampaignData } from "./campaign";

export enum CharacterRace {
	Dragonborn = "Dragonborn",
	Dwarf = "Dwarf",
	Elf = "Elf",
	Gnome = "Gnome",
	HalfElf = "HalfElf",
	Halfling = "Halfling",
	HalfOrc = "HalfOrc",
	Human = "Human",
	Tiefling = "Tiefling"
}

export enum CharacterClass {
	Barbarian = "Barbarian",
	Bard = "Bard",
	Cleric = "Cleric",
	Druid = "Druid",
	Fighter = "Fighter",
	Monk = "Monk",
	Paladin = "Paladin",
	Ranger = "Ranger",
	Rogue = "Rogue",
	Sorcerer = "Sorcerer",
	Warlock = "Warlock",
	Wizard = "Wizard"
}

export interface ICharacterData {
	Id?: number;
	Name: string;
	Level?: number;
	Race?: CharacterRace;
	Class?: CharacterClass;
	MaxHealth?: number;
	ArmorClass?: number;
	Notes?: string;

	Campaigns?: ICampaignData[];
}
