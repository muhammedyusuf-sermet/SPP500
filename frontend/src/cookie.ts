import { API_URL } from "./config";
import Store from 'electron-store';

export class CookieManager {
	private constructor(){
	}

	private static store = new Store<string>();

	public static UserToken(cookieToCheck:string):string {
		// undefined if user is not logged in.
		var cookieVal = CookieManager.store.get(cookieToCheck);

		if (cookieVal === undefined)
			return cookieVal

		var request = require("request");
		var options = { method: 'GET',
			url: API_URL + '/verify',
			timeout: 2000,
			headers:
			{
				'Cache-Control': 'no-cache',
				'Content-Type': 'application/json',
				'Authorization': cookieVal
			},
			body: {},
			json: true
		};

		request(options, function (error: string, response: string, body: {status: number, message: string}) {
			if (error) {
				return undefined
			}
			else {
				return (body.status === 201) ? "Valid cookie" : undefined // Success
			}
		});
		return CookieManager.store.get(cookieToCheck);
	}

	public static SetStringCookie(name:string, data:string) {
		CookieManager.store.set(name, data);
	}

	public static RemoveCookie(name:string) {
		CookieManager.store.delete(name);
	}
}
