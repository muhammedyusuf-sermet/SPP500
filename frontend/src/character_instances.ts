import { CharacterRace, CharacterClass, ICharacterData } from "./character";


let characterExample: ICharacterData;
characterExample = {
	Id: 1,
	Name: "Character",
	Level: 3,
	Race: CharacterRace.Gnome,
	Class: CharacterClass.Paladin,
	MaxHealth: 23,
	ArmorClass: 12,
	Notes: 'aggressive',

	Campaigns: []
};

export const CharacterInstances: ICharacterData[] = new Array();

// Generate a dummy array of characters with different IDs
for (var i = 0; i < 12; i++) {
	let character = {} as ICharacterData;
	Object.assign(character, characterExample);
	character.Id = i + 1;
	CharacterInstances.push(character);
}