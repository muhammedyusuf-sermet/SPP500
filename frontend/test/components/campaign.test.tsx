import * as React from "react"
import * as nock from 'nock';
import { shallow, ShallowWrapper } from 'enzyme';
import {Campaign, ICampaignState} from '../../src/renderer/components/platform/pages/view_game_components/Campaign';
import * as CampaignInterface from '../../src/campaign';
import {API_URL} from '../../src/config'

jest.mock('../../src/cookie');

describe('Test the Campaign View Details', () => {
	let campaignInstance: ShallowWrapper<any, ICampaignState, Campaign>;

	beforeEach(() => {
		campaignInstance = shallow(<Campaign/>);
	})

	it('renders without crashing', () => {
		expect(campaignInstance).toBeDefined();
	});

	it('renders correctly when the page is loaded', () => {
		expect(campaignInstance).toMatchSnapshot();
	});

	it('should update the state variables when resetState function is called', () => {
		campaignInstance.instance().resetState();

		expect(campaignInstance.state('page')).toEqual(0);
	});

	it('should make an GET request to retrieve Campaigns when getPaginatedCampaigns function is called', () => {
		campaignInstance.instance().getPaginatedCampaigns(0);

		nock(API_URL)
		.get('/campaign/get/0/12')
		.reply(201, { status: 201, message: 'success', total: 1, content: [] as CampaignInterface.ICampaignState[] });
	});

	it('should return the total number of pages when getTotalPages function is called', () => {
		campaignInstance.instance().setState({ totalCampaigns: 24});
		let totalPages = campaignInstance.instance().getTotalPages();
		// Starts from 0
		expect(totalPages).toEqual(1);
	});
});
