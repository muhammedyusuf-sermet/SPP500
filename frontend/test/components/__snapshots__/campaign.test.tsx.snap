// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Campaign View Details renders correctly when the page is loaded 1`] = `
<div
  className="layout card-grid"
  id="view-campaigns-container"
>
  <Pagination
    getTotalPages={[Function]}
    onPageChange={[Function]}
  />
  <WithStyles(Grid)
    container={true}
    spacing={40}
  />
</div>
`;
